### ISSUE TITLE
[path_provider] `getApplicationDocumentsDirectory` crashes the app on Windows If `Documents` is missing

### ISSUE THREAD

**Issue by syleishere** [flutter/flutter#80712]
import 'package:path_provider/path_provider.dart';

      Directory config_dir = await getApplicationDocumentsDirectory();
      String config_path = config_dir.path;
      print("Path writeable appears to be $config_path");

If you remove USER/OneDrive/My Documents folder from windows 10, then execute the above as app starts, app will completely crash. Seems to search for USER/OneDrive/My Documents no matter what even if I move it to USER/My Documents to get it off the cloud. This is supposed to be a "secure" path we can write to files, not sure why this is in the cloud on microsoft to begin with with directory, must be a better app directory it can look for for storing sensitive files.

**Comment by TahaTesser** [comment]
Hi @syleishere 
`My Document` is from user -> `C:\Users\Taha\My Document`, `Onedrive` has no such folder, you moved it there or create 

Running your code sample points to correct `My Document` and there is no crash

<details> 
<summary>code sample</summary> 

```dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:path_provider/path_provider.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () async {
          Directory config_dir = await getApplicationDocumentsDirectory();
          String config_path = config_dir.path;
          print("Path writeable appears to be $config_path");
        },
      ),
    );
  }
}
``` 

</details>

```console
Launching lib\main.dart on Windows in debug mode...
Connecting to VM Service at ws://127.0.0.1:50610/d_8nJmKsmH0=/ws
flutter: Path writeable appears to be C:\Users\Taha\Documents
```

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.5, on Microsoft Windows [Version 10.0.19042.928], locale en-US)
    • Flutter version 2.0.5 at C:\Users\Taha\Code\flutter_stable
    • Framework revision adc687823a (3 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.9.31129.286
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.3
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.55.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.19042.928]
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 89.0.4389.128
    • Edge (web)                         • edge          • web-javascript • Microsoft Edge 89.0.774.77

! Doctor found issues in 1 category.
```
	
</details>


Can you please provide your `flutter doctor -v` and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) and `flutter run -d windows --verbose` logs running the minimal code sample?
Thank you

**Comment by syleishere** [comment]
I ran your minimal example, crashed to once I pressed on button to execute the code to print the Path.
Go into your Taha folder and delete the Documents directory, then try and run it and you should be able to crash the app.
Can just restore it from recycle bin afterwards and then it will work.

**Comment by syleishere** [comment]
At this point I am unsure if this is onedrive related, I tried deleting onedrive off my computer, but I decided to restore it to stop crashing my app atm with that folder missing. Even if I tried re-creating directories manually, there must be a registry key its pulling that path from that needs to be changed if onedrive is removed, just a pain really, but regardless curious if your now getting same issue even though you don't have onedrive installed.

**Comment by TahaTesser** [comment]
Hi @syleishere 
I can reproduce the crash if I delete `Documents` folder

<details> 
<summary>code sample</summary> 

```dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:path_provider/path_provider.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () async {
          Directory config_dir = await getApplicationDocumentsDirectory();
          String config_path = config_dir.path;
          print("Path writeable appears to be $config_path");
        },
      ),
    );
  }
}
``` 

</details>

<details> 
<summary>logs</summary> 

```console
[  +90 ms] Syncing files to device Windows...
[   +2 ms] <- reset
[   +1 ms] Compiling dart to kernel with 0 updated files
[   +3 ms] <- recompile package:stable_flutter/main.dart 83897973-63d4-477f-9658-d9f51d8c3ce2
[        ] <- 83897973-63d4-477f-9658-d9f51d8c3ce2
[ +109 ms] Updating files.
[   +1 ms] DevFS: Sync finished
[   +2 ms] Syncing files to device Windows... (completed in 120ms)
[   +1 ms] Synced 0.0MB.
[   +1 ms] <- accept
[   +5 ms] Connected to _flutterView/0x18d95747490.
[   +2 ms] Flutter run key commands.
[   +2 ms] r Hot reload. 
[   +4 ms] R Hot restart.
[   +1 ms] h Repeat this help message.
[   +1 ms] d Detach (terminate "flutter run" but leave application running).
[   +1 ms] c Clear the screen
[   +1 ms] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on Windows is available at: http://127.0.0.1:65434/nXoy87egw30=/
[   +1 ms]
           Flutter DevTools, a Flutter debugger and profiler, on Windows is available at: http://127.0.0.1:9102?uri=http%3A%2F%2F127.0.0.1%3A65434%2FnXoy87egw30%3D%2F  
[   +1 ms] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[+1351 ms] Service protocol connection closed.
[   +1 ms] Lost connection to device.
[   +5 ms] DevFS: Deleting filesystem on the device (file:///C:/Users/Taha/AppData/Local/Temp/stable_flutterfca43383/stable_flutter/)
[ +254 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[   +2 ms] "flutter run" took 13,269ms.
[  +90 ms] ensureAnalyticsSent: 87ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +3 ms] Shutdown hooks complete
[   +1 ms] exiting with code 0
```

</details>

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.5, on Microsoft Windows [Version 10.0.19042.928], locale en-US)
    • Flutter version 2.0.5 at C:\Users\Taha\Code\flutter_stable
    • Framework revision adc687823a (4 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.9.31129.286
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.3
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.55.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.928]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 89.0.4389.128
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 89.0.774.77

! Doctor found issues in 1 category.
```

```bash
[✓] Flutter (Channel master, 2.2.0-11.0.pre.174, on Microsoft Windows [Version 10.0.19042.928], locale en-US)
    • Flutter version 2.2.0-11.0.pre.174 at C:\Users\Taha\Code\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision eaa2cfdd3a (4 hours ago), 2021-04-19 23:39:02 -0400
    • Engine revision 60f12daf85
    • Dart version 2.14.0 (build 2.14.0-8.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.9.31129.286
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.3
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.55.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.928]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 89.0.4389.128
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 89.0.774.77

! Doctor found issues in 1 category.
```
	
</details>
